environment.go

```diff
@@ -307,6 +314,101 @@ func (e *envState) ClientConfig() *ClientConfig {
 	return &config
 }
 
+func (e *envState) GetSender(key SenderKey) (send.Sender, error) {
```

Comment:
my inclination is to create/instantiate the senders on init (like we do with the queues, and then just have these pull them out of a map.

Reply:
Can you comment on the thread safety of grip senders? I'm most worried about email, which I expect to be a popular one, but also because the sender locks the client.

Further more, there is the issue of the DB config, if we initialise the senders on start-up, they won't receive changes to config w/o restarting, which I think kind of defeats a strong advantage of that project (i.e. changing settings w/o restarting everything.)

Reply:
grip senders are thread safe. 

There are a lot of settings that currently require restarts (e.g. amboy worker pool size,) most everything else is passed a settings object at start up time (web servers/etc.) and I think that's fine. The real value of the config-in-db-project was avoiding needing to distribute/chef the config file; restarts are easy.